river/command/filter.zig

```diff
@@ -33,13 +33,24 @@ pub fn floatFilterAdd(
     args: []const [:0]const u8,
     out: *?[]const u8,
 ) Error!void {
-    if (args.len < 2) return Error.NotEnoughArguments;
-    if (args.len > 2) return Error.TooManyArguments;
+    if (args.len < 3) return Error.NotEnoughArguments;
+    if (args.len > 3) return Error.TooManyArguments;
 
-    const gop = try server.config.float_filter.getOrPut(util.gpa, args[1]);
+    var kind = args[1];
+    var map = &server.config.float_filter_app_ids;
```

Comment:
I don't really like this, but I couldn't get it to work in any other way. Ideally, I'd like something like this:

```zig
var map = switch (kind) {
    "app-id" => ...,
    "title" => ...,
    else => return Error.UnknownOption,
};
```

but it seems like switching on strings is not supported (yet?).

Having the map be uninitialized would be fine as well, but that doesn't compile (I tried `var map;`). Maybe there is some Zig-foo I am missing to make this work @ifreund ?

Reply:
something like this would work:

```zig

Reply:
Oh that is much better. Thanks!